server/src/main/java/org/apache/druid/server/metrics/QueryCountStatsMonitor.java

```diff
@@ -72,9 +75,11 @@ public boolean doMonitor(ServiceEmitter emitter)
       }
     }
 
-    long pendingQueries = this.mergeBufferPool.getPendingRequests();
-    emitter.emit(builder.setMetric("mergeBuffer/pendingRequests", pendingQueries));
+    if (emitMergeBufferPendingRequests) {
+      long pendingQueries = this.mergeBufferPool.getPendingRequests();
+      emitter.emit(builder.setMetric("mergeBuffer/pendingRequests", pendingQueries));
+    }
+
```

Comment:
Why is this required? Isn't it cleaner to not emit the metric from the GroupByStatsMonitor at all and keep it as is?

Reply:
Ideally, all the merge buffer metrics should be reported from the new monitor. This change is to ensure backward compatibility.

Reply:
But we don't have any plans of removing this code right? Which makes me wonder if the additional complexity is worth it. The situation is that we have had a metric in past, which was emitted without a monitor, and we are not planning on modifying the behavior (LMK if this part is not true, which will make the comment null & void). So it doesn't seem correct that we are emitting this from the monitor as well for correctness given that we also have weird artefacts of this metric in QueryCountStatsMonitor which would not be removed.

Reply:
> The situation is that we have had a metric in past, which was emitted without a monitor, and we are not planning on modifying the behavior 

This isn't right, the metric was emitted from a monitor earlier (QueryCountStatsMonitor.java). 

We can remove the metric from the old monitor and the corresponding code in the next release?